testsuite/MDAnalysisTests/analysis/test_contacts.py

```diff
@@ -171,16 +171,64 @@ class TestContacts(object):
     def universe():
         return mda.Universe(PSF, DCD)
 
-    def _run_Contacts(self, universe, start=None, stop=None, step=None, **kwargs):
+    def _run_Contacts(self, universe, sel_acidic=None, sel_basic=None, start=None, stop=None, step=None, **kwargs):
         acidic = universe.select_atoms(self.sel_acidic)
         basic = universe.select_atoms(self.sel_basic)
+        if sel_acidic is None:
+            sel_acidic = self.sel_acidic
+        if sel_basic is None:
+            sel_basic = self.sel_basic
+
         return contacts.Contacts(
             universe,
-            select=(self.sel_acidic, self.sel_basic),
+            select=(sel_acidic, sel_basic),
             refgroup=(acidic, basic),
             radius=6.0,
             **kwargs).run(start=start, stop=stop, step=step)
 
+    def test_select_types(self, universe):
+        """Test if Contact can take both string and AtomGroup as selections.
+        """
+        acidic = universe.select_atoms(self.sel_acidic)
+        basic = universe.select_atoms(self.sel_basic)
+
+        with_string = self._run_Contacts(universe)
+        with_atomgroup = self._run_Contacts(universe, sel_acidic=acidic, sel_basic=basic)
+       
+        assert_equal(with_atomgroup.grA, with_string.grA)
+        assert_equal(with_atomgroup.grB, with_string.grB) 
+
+    @pytest.mark.xfail
```

Comment:
rather than using `xfail` it would be much better to use the `with pytest.raises()` construct with a match to the error message you are expecting (`select_error_message`).

Reply:
This is indeed a better approach! I updated the testing code to use `pytest.raises()`, and `pytest.mark.parametrize` to combine all expected failure tests in one function. As for referring to the `select_error_message`, I updated `contacts.Contacts` class to define it as a static attribute, and referred to it in `test_contacts.py` using `with pytest.raises(TypeError, match=contacts.Contacts.select_error_message)`. Is that okay, or is there a better way?